src/util/Exception.h

```diff
@@ -227,28 +242,30 @@ class Exception {
   //! Set error details
   void setErrorDetails(const string& errorDetails) {
     _errorDetails = errorDetails;
-    _errorDetailsNoFileAndLines = _errorDetailsNoFileAndLines;
   }
 
   //! Get error Code
-  int getErrorCode() const { return _errorCode; }
+  int getErrorCode() const noexcept { return _errorCode; }
 
   //! Get error message pertaining to code
-  string getErrorMessage() const { return errorCodeAsString(_errorCode); }
+  string getErrorMessage() const noexcept {
+    return errorCodeAsString(_errorCode);
```

Comment:
In theory this might throw (involves a `std::string` constructor). But this still might be intended, because in those rare cases we probably cannot save our software at all anymore.

Reply:
Yes I think this would be a very weird error for which crashing might be the best thing to do.